sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/ExceptionEventData.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.data;
+
+import io.opentelemetry.api.common.Attributes;
+
+public interface ExceptionEventData extends EventData {
```

Comment:
I believe this can be package private.

Reply:
Oh nvm. It has to be public so that processors / exporters can check `instanceof ExceptionEventData`. This seems reasonable to me too. 

If we wanted to be cautious we could put this interface in `:sdk-extensions:tracing-incubator` so users have to add a dependency on an unstable artifact in order to use it.

Reply:
The SDK would need to know of the interface in order for `Span#recordException` to add an event that implements it, wouldn't it?

Reply:
What about putting this in an `internal` package for now, as we explicitly document anything in `internal` as being unstable. You would be free to use it, of course, accepting the risk of it changing in the future.

Reply:
> The SDK would need to know of the interface in order for Span#recordException to add an event that implements it, wouldn't it?

Yes, it could place an implementation dependency on `:sdk-extensions:tracing-incubator`.

> What about putting this in an internal package for now, as we explicitly document anything in internal as being unstable. 

That would work too, but probably better to not promote using internal classes.

Reply:
I don't want to promote it...but @HaloFour could try it out while we figure out spec issues.

Reply:
Also, it could be seen as an implementation detail for now, with future promotion out of the `internal` packaging.

Reply:
That's definitely reasonable - I don't have a strong preference.

Reply:
If the `internal` package is sufficient I can move to that.

Otherwise how would you feel about having the `ExceptionEventData` interface in `tracing-incubator` and SDK picks it up if it detects it on the class path?

Reply:
I'd be fine with that as well.

Reply:
Finished pass at moving the `ExceptionEventData` interface and implementation to `tracing-incubator`.  Added a helper class that looks for it on the classpath and will use the `create` static method to create instances, otherwise it defaults to the existing behavior.  Not terribly thrilled with the manual use of reflection but it seems that `MethodHandles` and `LambdaMetaFactory` aren't available due to the target Android API version and desugaring seems to only apply to them when used with `invokedynamic`, but hopefully given this code is related to exceptions the overhead of invoking the method is relatively a drop in the bucket.

Also made the implementation of `ExceptionEventData` defer creating of the common attributes until the `getAttributes` method is called and the result is memoized, so existing exporters should work as expected if using this implementation.  That behavior is certainly up for discussion.

Reply:
> Do users need to opt out?

Not necessarily, just thought it was safer to be able to test it in isolation.

I can get the PR together to move these classes back to `:sdk:tracing` under an `internal` package.